src/mavedb/routers/authorization.py

```diff
@@ -0,0 +1,73 @@
+import logging
+from enum import Enum
+
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.orm import Session
+from typing import Union, Optional
+
+from mavedb import deps
+from mavedb.lib.authentication import get_current_user, UserData
+from mavedb.lib.permissions import has_permission, Action
+from mavedb.lib.logging import LoggedRoute
+from mavedb.lib.logging.context import logging_context, save_to_logging_context
+from mavedb.models.experiment import Experiment
+from mavedb.models.experiment_set import ExperimentSet
+from mavedb.models.score_set import ScoreSet
+
+router = APIRouter(
+    prefix="/api/v1",
+    tags=["authorizations"],
+    responses={404: {"description": "Not found"}},
+    route_class=LoggedRoute,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class ModelName(str, Enum):
+    experiment = "experiment"
+    experiment_set = "experiment-set"
+    score_set = "score-set"
+
+
+@router.get(
+    "/user-is-authorized/{model_name}/{urn}/{action}",
+    status_code=200,
+    response_model=bool
+)
+async def check_authorization(
+    *,
+    model_name: str,
+    urn: str,
+    action: str,
```

Comment:
Here if we define the action like
```suggestion
    action: Action,
```
plus change the definition of the Action enum in the file `lib/permissions.py` to look like
```
class Action(Enum):
    READ = 'read'
    UPDATE = 'update'
    DELETE = 'delete'
    ADD_EXPERIMENT = 'add_experiment'
    ADD_SCORE_SET = 'add_score_set'
    SET_SCORES = 'set_scores'
    ADD_ROLE = 'add_role'
    PUBLISH = 'publish'
```
Fastapi will automatically validate the API call to enforce one of those strings as the action, and raise a `422` error if the action is not one of those options. For an example of that, see this page: https://fastapi.tiangolo.com/tutorial/path-params/#predefined-values